Subject: Re: IDE not found on Performa due to interrupt breakage
From: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Patch-mainline: 2.6.19

On Thu, 2006-09-21 at 13:50 +0200, Olaf Hering wrote:
> The IDE controller is not usable on a Performa 6400 with 2.6.18:
>
> <6>hda: Enabling MultiWord DMA 2
> <4>ide0: Disabled unable to get IRQ 13.
> <6>ide0: failed to initialize IDE interface

 .../...

Looks like a workaround for bogus OF bitrotted... What about this patch:

Signed-off-by: Olaf Hering <olh@suse.de>
---
 drivers/ide/ppc/pmac.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.18-rc6/drivers/ide/ppc/pmac.c
===================================================================
--- linux-2.6.18-rc6.orig/drivers/ide/ppc/pmac.c
+++ linux-2.6.18-rc6/drivers/ide/ppc/pmac.c
@@ -1326,7 +1326,7 @@ pmac_ide_macio_attach(struct macio_dev *
 	if (macio_irq_count(mdev) == 0) {
 		printk(KERN_WARNING "ide%d: no intrs for device %s, using 13\n",
 			i, mdev->ofdev.node->full_name);
-		irq = 13;
+		irq = irq_create_mapping(NULL, 13);
 	} else
 		irq = macio_irq(mdev, 0);
 
